torchvision/datasets/kinetics.py

```diff
@@ -74,6 +148,89 @@ def __init__(self, root, frames_per_clip, step_between_clips=1, frame_rate=None,
+
+        Raises:
+            RuntimeError: if download folder exists, break to prevent downloading entire dataset again.
+        """
+        if path.exists(self.split_folder):
+            raise RuntimeError(
+                f"The directory {self.split_folder} already exists. If you want to re-download or re-extract the images, "
+                f"delete the directory."
+            )
+        # check that the assignment was made properly
+        tar_path = path.join(self.root, "tars")
+        annotation_path = path.join(self.root, "annotations")
+        file_list_path = path.join(self.root, "files")
+
+        split_url = self._TAR_URLS[self.num_classes].format(split=self.split)
+        split_url_filepath = path.join(file_list_path, path.basename(split_url))
+        if not check_integrity(split_url_filepath):
+            download_url(split_url, file_list_path)
+        list_video_urls = open(split_url_filepath, "r")
+
+        if not check_integrity(path.join(annotation_path, f"{self.split}.csv")):
+            download_url(self._ANNOTATION_URLS[self.num_classes].format(split=self.split), annotation_path)
+        self.annotations = path.join(annotation_path, f"{self.split}.csv")
+
+        if self.num_download_workers == 1:
+            for line in list_video_urls.readlines():
+                line = str(line).replace("\n", "")
+                download_and_extract_archive(line, tar_path, self.split_folder)
+        else:
+            part = partial(_dl_wrap, tar_path, self.split_folder)
+            lines = [str(line).replace("\n", "") for line in list_video_urls.readlines()]
+            poolproc = Pool(self.num_download_workers)
+            poolproc.map(part, lines)
+
+    def _make_ds_structure(self):
+        """move videos from
+        split_folder/
+            ├── clip1.avi
+            ├── clip2.avi
+
+        to the correct format as described below:
+        split_folder/
+            ├── class1
+            │   ├── clip1.avi
+
+        """
+        file_tmp = "{ytid}_{start:06}_{end:06}.mp4"
+        with open(self.annotations) as csvfile:
+            reader = csv.DictReader(csvfile)
+            for row in reader:
+                f = file_tmp.format(
+                    ytid=row["youtube_id"],
+                    start=int(row["time_start"]),
+                    end=int(row["time_end"]),
+                )
+                label = (
+                    row["label"]
+                    .replace(" ", "_")
+                    .replace("'", "")
+                    .replace("(", "")
+                    .replace(")", "")
+                )
+                os.makedirs(path.join(self.split_folder, label), exist_ok=True)
+                existing_file = path.join(self.split_folder, f)
+                if path.isfile(existing_file):
+                    os.replace(
+                        existing_file, path.join(self.split_folder, label, f),
+                    )
```

Comment:
I remember asking myself this before, but I can't find the comment if I added one: how can a file be pre-existing here? Given that we always start from a clean folder, the only possibility is a duplicate entry in the annotations file. If this happens, an error is warranted, no?

Reply:
Actually, what happens is a missing entry; there are several videos missing that still exist in official annotations for their "backwards compatibility". I can raise an error, but we have to have this otherwise it breaks.